plugins/scaffolder-actions/scaffolder-backend-module-aws/src/actions/s3/cp.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2021 Larder Software Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { createTemplateAction } from '@backstage/plugin-scaffolder-backend';
+import { S3Client, PutObjectCommand, S3ClientConfig } from '@aws-sdk/client-s3';
+import fs, { createReadStream } from 'fs-extra';
+import { resolveSafeChildPath } from '@backstage/backend-common';
+import glob from 'glob';
+import { CredentialProvider } from '@aws-sdk/types';
+import { assertError } from '@backstage/errors';
+
+export function createAwsS3CpAction(options?: {
+  credentials?: CredentialProvider;
+}) {
+  return createTemplateAction<{
+    path?: string;
+    prefix?: string;
+    bucket: string;
+    region: string;
+  }>({
```

Comment:
We can extend this by adding
```
      endpoint?: string
      s3ForcePathStyle?: boolean
```
to support a ton of "AWS S3 compatible" storages. These can be passed into the config with those names so it's a quick addition.

Can be done in the next iteration if you don't want to tackle it now.

Reply:
Added them as you said it was a super easy addition